storage/flux/table.gen.go.tmpl

```diff
@@ -702,24 +703,87 @@ func (t *{{.name}}GroupTable) Do(f func(flux.ColReader) error) error {
 }
 
 func (t *{{.name}}GroupTable) advance() bool {
-RETRY:
-	a := t.cur.Next()
-	l := a.Len()
-	if l == 0 {
-		if t.advanceCursor() {
-			goto RETRY
-		}
-
+	if t.cur == nil {
+		// For group aggregates, we will try to get all the series and all table buffers within those series
+		// all at once and merge them into one row when this advance() function is first called.
+		// At the end of this process, t.advanceCursor() already returns false and t.cur becomes nil.
+		// But we still need to return true to indicate that there is data to be returned.
+		// The second time when we call this advance(), t.cur is already nil, so we directly return false.
 		return false
 	}
+	var a *cursors.{{.Name}}Array
+	var l int
+	for {
+		a = t.cur.Next()
+		l = a.Len()
+		if l > 0 {
+			break
+		}
+		if !t.advanceCursor() {
+			return false
+		}
+	}
 
-	// Retrieve the buffer for the data to avoid allocating
-	// additional slices. If the buffer is still being used
-	// because the references were retained, then we will
-	// allocate a new buffer.
-	cr := t.allocateBuffer(l)
-	cr.cols[timeColIdx] = arrow.NewInt(a.Timestamps, t.alloc)
-	cr.cols[valueColIdx] = t.toArrowBuffer(a.Values)
+	// handle the group without aggregate case
+	if t.gc.Aggregate() == nil {
+		// Retrieve the buffer for the data to avoid allocating
+		// additional slices. If the buffer is still being used
+		// because the references were retained, then we will
+		// allocate a new buffer.
+		cr := t.allocateBuffer(l)
+		cr.cols[timeColIdx] = arrow.NewInt(a.Timestamps, t.alloc)
+		cr.cols[valueColIdx] = t.toArrowBuffer(a.Values)
+		t.appendTags(cr)
+		t.appendBounds(cr)
+		return true
+	}
+
+	// handle the group with aggregate case
+	var value {{.Type}}
+	// For group count, sum, min, and max, the timestamp here is always math.MaxInt64.
+	// their final result does not contain _time, so this timestamp value can be anything
+	// and it won't matter.
+	// For group first, we need to assign the initial value to math.MaxInt64 so
+	// we can find the row with the smallest timestamp.
+	// Do not worry about data with math.MaxInt64 as its real timestamp.
+	// In OSS we require a |> range() call in the query and a math.MaxInt64 timestamp
+	// cannot make it through.
```

Comment:
Yeah, this is the right kind of comment: we all have this common knowledge, but this code may need to be modified by someone without this common knowledge.